src/ch/blinkenlights/android/vanilla/PlaybackService.java

```diff
@@ -979,6 +1002,10 @@ private void processNewState(int oldState, int state)
 			mTimeline.setShuffleMode(shuffleMode(state));
 		if ((toggled & MASK_FINISH) != 0)
 			mTimeline.setFinishAction(finishAction(state));
+
+		if((toggled & FLAG_DUCKING) != 0) {
+			mMediaPlayer.setIsDucking((state & FLAG_DUCKING) != 0);
+		}
```

Comment:
They are swapped after we moved to the next song in the queue:
Playing a Song causes VanillaMusic to:
* Set the song as the source in mMediaPlayer
* Set the NEXT song in the queue as source in mPreparedMediaPlayer

We can then call mMediaPlayer.setNextMediaPlayer(mPreparedMediaPlayer) to tell android which song it should play after the current song finished: This is needed for gapless playback.

The actual swapping of the two objects happens in processSong();
mPreparedMediaPlayer.isPlaying() will be true if androids MediaPlayer moved to the next song:
This causes the code to simply swap mMediaPlayer & mPreparedMediaPlayer.